Allow custom ignored namespaces in backtrace processor

adam-ja opened the pull request:
The backtrace processor adds the file, line, class, and function
responsible for triggering a log to the event extras.

By default, classes in the backtrace under the `Zend\Log` namespace are
ignored so that we can see the actual application code that called the
`log()`.

However, there might also be cases when we want to ignore other
namespaces so that we get the true origin. The example I came across
that prompted this patch was `Psr\Log`, which appears when the
`PsrLoggerAdapter` wraps the underlying Zend Logger.

This patch allows us to add our own custom ignored namespaces in
addition to the default `Zend\Log`.

Commit:
Allow custom ignored namespaces in backtrace processor

The backtrace processor adds the file, line, class, and function
responsible for triggering a log to the event extras.

By default, classes in the backtrace under the Zend\Log namespace are
ignored so that we can see the actual application code that called the
log().

However, there might also be cases when we want to ignore other
namespaces so that we get the true origin. The example I came across
that prompted this patch was Psr\Log, which appears when the
PsrLoggerAdapter wraps the underlying Zend Logger.

This patch allows us to add our own custom ignored namespaces in
addition to the default Zend\Log.

weierophinney commented:
This looks like a great feature!

Can you please also add documentation (under the `doc/book/` tree) detailing the new methods and options?

Thanks!

weierophinney committed:
Merge branch 'hotfix/87'

Close #87

adam-ja committed:
Docs for ignored namespaces in Backtrace Processor

adam-ja commented:
Docs added - sorry for the delay.

There's really nothing to say about `getIgnoredNamespaces()` as there's probably not a great deal of use for it beyond testing/debugging.

Xerkus commented on src/Processor/Backtrace.php:
```diff
@@ -59,6 +59,31 @@ public function process(array $event)
+     * @param string $namespace
+     * @return self
+     */
+    public function addIgnoredNamespace($namespace)
```
Can it be changed to constructor injection?
I see no reason to change value after processor was created, as new namespaces to ignore should really be added by adding more configured processors

Commit:
Made ignoredNamespaces a constructor option

Ignored namespaces are not expected to change once a processor is
created so it makes more sense to have this as a constructor option
rather than a separate function that allows the processor to be
reconfigured after creation.

Commit:
Fix documentation for configuring processor via $logger->addProcessor()

michalbundyra commented:
Thanks, @adam-ja!
